### PR TITLE
Unblock S.IO.Ports and S.IO.Pipes on Aot.

### PR DESCRIPTION
**Pull request by AtsushiKan**
More stray P/Invokes...

### CODE REVIEW

File: src/System.IO.Ports/tests/SerialPort/DosDevices.cs [hunk 1]
```diff
@@ -139,7 +139,7 @@ private static char[] CallQueryDosDevice(string name, out int dataSize)
         public const int ERROR_INSUFFICIENT_BUFFER = 122;
         public const int ERROR_MORE_DATA = 234;
 
-        [DllImport("Kernel32.dll", SetLastError = true, CharSet = CharSet.Auto)]
+        [DllImport("Kernel32.dll", SetLastError = true, EntryPoint = "QueryDosDeviceW", CharSet = CharSet.Unicode)]
```
Comment (danmosemsft) [inline src/System.IO.Ports/tests/SerialPort/DosDevices.cs]: Why is EntryPoint required? We often don't add it, I think. The interop infra tries the method name, then adding W.

Comment (AtsushiKan) [inline src/System.IO.Ports/tests/SerialPort/DosDevices.cs]: The A/W probing doesn't happen on .NET Native. This is exactly was causing Ports to die on startup.

File: src/System.IO.Pipes/tests/Interop.Windows.cs [hunk 2]
```diff
@@ -65,31 +63,8 @@ internal static bool TryGetNumberOfServerInstances(SafePipeHandle handle, out in
...
-                return true;
-            }
-
-            hostName = "";
-            return false;
-        }
-
-        #endregion
+        // @todo: These are called by some Unix-specific tests. Those tests should really be split out into
+        // partial classes and included only in Unix builds.
```
Comment (ianhays) [inline src/System.IO.Pipes/tests/Interop.Windows.cs]: Is this TODO not done? It looked like it was, and if not then lets do it :)

File: src/System.IO.Pipes/tests/Interop.Unix.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+                hostName = Marshal.PtrToStringAnsi((IntPtr)name);
+                return true;
+            }
+
+            hostName = "";
+            return false;
+        }
+
+        // @todo: These are called by some Windows-specific tests. Those tests should really be split out into
+        // partial classes and included only in Windows builds.
```
Comment (ianhays) [inline src/System.IO.Pipes/tests/Interop.Unix.cs]: Is this TODO not done? It looked like it was, and if not then lets do it :)

File: src/System.IO.Pipes/tests/Interop.Windows.cs (hunk 2 above)
Comment (AtsushiKan) [inline src/System.IO.Pipes/tests/Interop.Windows.cs]: No, the goal of the PR is to get the test unblocked on AoT as quickly as possible. The todo's are just nice-to-do cleanup, but they're definitely a pri-3 issue.

**Comment by willdean** [comment]
This was a serious regression, as it actually stops PortHelper,GetPorts finding any ports at all - this means that all tests which use hardware (i.e. the vast majority) get skipped.  This doesn't show up on CI because we skip most of the tests anyway.

If MS is going to build/ship SerialPort at all, I'm afraid you're going to need to somehow invest in the hardware support to test against some kind of hardware.

I think it's a charset issue, but I'm still trying to debug it (heh, see other recent threads...).   I'll put in a PR when I know what's happening.

**Pull request merged**